loader/0002-quilt.mod.json.md

```diff
@@ -0,0 +1,388 @@
+```json
+{
+    "name": "Perfectly Awesome License v1.0",
+    "id": "PAL-1.0",
+    "url": "https://theperfectlyawesomelicense.com/",
+    "description": "This license does things and stuff and says that you can do things and stuff too!"
+}
+```
+The `"description"` field is optional.
+
+#### The `icon` field
+| Type          | Required |
+|---------------|----------|
+| Object/String | False    |
+
+One or more paths to a square .PNG file. If an object is provided, the keys must be the resolution of the corresponding file. For example:
+```json
+"icon": {
+    "32": "path/to/icon32.png",
+    "64": "path/to/icon64.png",
+    "4096": "path/to/icon4096.png"
+}
+```
+
+## The `mixin` field
+| Type         | Required |
+|--------------|----------|
+| Array/String | False    |
+
+A single or array of paths to mixin configuration files relative to the root of the mod JAR.
+
+## The `access_widener` field
+| Type         | Required |
+|--------------|----------|
+| Array/String | False    |
+
+A single or array of paths to accesswidener files relative to the root of the mod JAR.
+
+## The `minecraft` field
+| Type   | Required |
+|--------|----------|
+| Object | False    |
+
+Contains flags and options related to Minecraft specifically.
+
+### The `environment` field
+| Type   | Required |
+|--------|----------|
+| String | False    |
+
+Defines the environment(s) that this mod should be loaded on. Valid values are:
+* `"*"` — All environments (default)
+* `"client"` — The physical client
+* `"server"` — The dedicated server
+
+## Custom Elements
+In addition to the defined elements above, mods and libraries will be able to add their own elements to the quilt mod file. Mods will be expected to define up to one top-level element corresponding to their mod id. The element can be of any type, so that mods can define either a single value, array of values, or a sub-object.
+
+## Example
+An example quilt.mod.json:
+```json
+{
+    "schema_version": 1,
+    "quilt_loader": {
+        "group_id": "org.quiltmc",
+        "mod_id": "example_mod",
+        "version": "1.0.0",
+        "entrypoints": {
+            "main": [
+                "org.quiltmc.example_mod.impl.ExampleMod",
+                "org.quiltmc.example_mod.impl.ExampleModNetworking"
+            ],
+            // Since we only have a single client endpoint, no array is needed.
+            "client": "org.quiltmc.example_mod.impl.client.ExampleModClient",
+        },
+        "depends": [
+            "quilt_networking_api",
+            "quilt_rendering_api",
+            {
+                "id": "modmenu",
+                "environment": "client"
+            }
+        ],
+        "breaks": [
+            {
+                "id": "sodium",
+                "versions": "*",
+                "reason": "Sodium does not implement the Quilt Rendering API."
+            },
+            {
+                "id": "some_random_library",
+                "versions": [
```

Comment:
Uuh, little problem: there's nothing that defines a `versions` field that accepts an array of id-less dependency objects and/or version specifiers in this specification. Perhaps something about them should be written